subscriber.go

```diff
@@ -626,9 +669,24 @@ func (h *handler) handleAsync(ctx context.Context, nextCid cid.Cid, ss ipld.Node
 // handle processes a message from the peer that the handler is responsible for.
 // The caller is responsible for ensuring that this is called while h.syncMutex
 // is locked.
-func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wrapSel, updateLatest bool, syncer Syncer) error {
+func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wrapSel, updateLatest bool, syncer Syncer, hook BlockHookFunc, scopedBlockHookMutex *sync.RWMutex, scopedBlockHook map[peer.ID]BlockHookFunc) error {
 	log := log.With("cid", nextCid, "peer", h.peerID)
 
+	// This is not set to nil so we can get a pointer.
+	syncedCids := []cid.Cid{}
+	hook = WrapBlockHookWithSyncedCidTracker(&syncedCids, hook)
+	scopedBlockHookMutex.Lock()
+	scopedBlockHook[h.peerID] = hook
+	scopedBlockHookMutex.Unlock()
```

Comment:
Change these to `h.subscriber.scopedBlockHookMutex` and `h.subscriber.scopedBlockHook`